src/playercards/CardsThatSealTokens.ttslua

```
@@ -0,0 +1,154 @@
+  elseif SHOW_MULTI_RELEASE then
+    self.addContextMenuItem("Release " .. SHOW_MULTI_RELEASE .. " token(s)", function(playerColor)
+      if SHOW_MULTI_RELEASE >= #sealedTokens then
+        for i = 1, SHOW_MULTI_RELEASE do
+          releaseOneToken(playerColor)
+        end
+      else
+        printToColor("Not enough " .. name .. " tokens sealed.", playerColor)
+      end
+    end)
+  else
+    self.addContextMenuItem("Release token(s)", releaseTokens)
+  end
+
+  -- main context menu options to seal tokens
+  for _, map in pairs(ID_URL_MAP) do
+    if (VALID_TOKENS[map.name] ~= nil) or (SHOW_READ_BAG and tokensInBag[map.name] and not INVALID_TOKENS[map.name]) then
+      if not SHOW_MULTI_SEAL then
+        self.addContextMenuItem("Seal " .. map.name, function(playerColor)
+          sealToken(map.name, playerColor)
+        end)
+      else
+        self.addContextMenuItem("Seal " .. SHOW_MULTI_SEAL .. " " .. map.name, function(playerColor)
+          readBag()
+          local allowed = true
+          local notFound
+
+          for name, _ in pairs(VALID_TOKENS) do
+            if (tokensInBag[name] or 0) < SHOW_MULTI_SEAL then
+              allowed = false
+              notFound = name
+            end
+          end
+
+          if allowed then
+            for i = 1, SHOW_MULTI_SEAL do
+              sealToken(map.name, playerColor)
+            end
+          else
+            printToColor("Not enough " .. notFound .. " tokens in the chaos bag.", playerColor)
+          end
+        end)
+      end
+    end
+  end
+end
+
+-- generates a list of chaos tokens that is in the chaos bag
+function readBag()
+  local chaosbag = Global.call("findChaosBag")
+  tokensInBag = {}
+
+  for _, token in ipairs(chaosbag.getObjects()) do
+    tokensInBag[token.name] = (tokensInBag[token.name] or 0) + 1
+  end
+end
+
+-- seals the named token on this card
+function sealToken(name, playerColor)
+  if not Global.call("canTouchChaosTokens") then return end
```

Add an explanatory message here rather than just failing silently.  Apply to all

---

This message is already include in the called function in Global:
`broadcastToAll("Someone is searching the chaos bag, can't touch the tokens.", "Red")`